homeassistant/components/tp_link_eap/device_tracker.py

```diff
@@ -0,0 +1,114 @@
+"""Support for extracting client link form a TP-Link EAP AP."""
+import logging
+import re
+
+from pexpect import pxssh
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    DOMAIN,
+    PLATFORM_SCHEMA,
+    DeviceScanner,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_INTERFACES, DEFAULT_INTERFACES, DEFAULT_PORT
+
+_LOGGER = logging.getLogger(__name__)
+
+MAC_RE = re.compile("([A-Fa-f0-9]{2}:?){6}")
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+        vol.Optional(CONF_INTERFACES, default=DEFAULT_INTERFACES): cv.ensure_list,
+    }
+)
+
+
+def get_scanner(hass, config):
+    """Validate the configuration and return a TP-Link EAP scanner."""
+    return TpLinkEapDeviceScanner(config[DOMAIN])
+
+
+class TpLinkEapDeviceScanner(DeviceScanner):
+    """This class scans for devices using a SSH connection to a TP-Link EAP AP."""
+
+    exclude = []
+
+    def __init__(self, config):
+        """Initialize the scanner."""
+        self.last_results = []
+
+        self.host = config[CONF_HOST]
+        self.user = config[CONF_USERNAME]
+        self.password = config[CONF_PASSWORD]
+        self.port = config[CONF_PORT]
+        self.interfaces = config[CONF_INTERFACES]
+        _LOGGER.debug(
+            "Scanner initialized for TP-Link EAP %s with interfaces %s",
+            self.host,
+            self.interfaces,
+        )
+
+    def scan_devices(self):
+        """Scan for devices."""
+        self._update_info()
+
+        _LOGGER.debug("Scan last results %s", self.last_results)
+
+        return self.last_results
+
+    def get_device_name(self, device):
+        """Get the name of a device."""
+        return None
+
+    def get_extra_attributes(self, device):
+        """Get the extra attributes of a device."""
+        return {}
+
+    def _update_info(self):
+        """Scan the network for devices.
+
+        Returns boolean if scanning successful.
+        """
+        _LOGGER.debug("Scanning...")
+
+        session = pxssh.pxssh(echo=False, encoding="utf-8")
```

Comment:
Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices

Reply:
Sorry, I basically copy&pasted an existing device tracker and did not realize that it was not allowed anymore for new component.